src/olympia/addons/models.py

```diff
@@ -487,12 +487,15 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
-        # Reclaim GUID from deleted add-on.
-        try:
-            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
-            old_guid_addon.update(guid=None)
-        except ObjectDoesNotExist:
-            old_guid_addon = None
+        guid = data.get('guid')
```

Comment:
Usually, a `CharField` should not be nullable (and should only be the empty string), see https://docs.djangoproject.com/es/1.9/ref/models/fields/#django.db.models.Field.null

However, historically, we've always been using nullable character fields, so yes indeed I could write it this way to be more explicit, but on the other hand, I'd rather not rely on this "bad behavior" still being used in the future, just for the sake of explicitness.